bitcoin_lib/sources/encoding.move

```diff
@@ -227,6 +227,36 @@ public fun parse_btc_sig(full_sig_from_stack: &mut vector<u8>): (vector<u8>, u8)
     (r_and_s_bytes, sighash_flag)
 }
 
+/// Fomat raw singature (r, s) to btc ECDSA format,
+/// Ref: https://learnmeabitcoin.com/technical/keys/signature/
+public fun der_encode_signature(signature: vector<u8>, signature_hash_type: u8): vector<u8> {
```

Comment:
i am pretty sure we implemented something similar for the bitcion_executor. Maybe we can just move that one here